solver/src/main/java/org/chocosolver/solver/constraints/nary/nvalue/PropNValue.java

```diff
@@ -34,7 +35,7 @@
     private IntVar nValue;
     private final int n;
     private int[] concernedValues;
-    private int[] witness;
+    private IStateInt[] witness;
```

Comment:
hmmm.. this shouldn't be backtrackable !
This is a regression

Reply:
Making witness array backtrackable gives a quicker propagator (at least for the corresponding test) and ease backtracking.

Reply:
that's a nonsense: witness (or watch literal) are natively backtrackable.
Indeed, a witness points to a supported value, which is always true on backtrack, so no need to make backtrackable.
If that is quicker, you should think of why it is, because since you are doing less operations, it shouldn't be quicker.

Reply:
You are right. It was the getting back to -1 for the witness that was posing problems. Now, it should be good, and the propagator is indeed faster with an int[] array